nova/network/neutronv2/api.py

```diff
@@ -1314,6 +1314,18 @@ def bind_ports_to_host(self, context, instance, host,
             if resp:
                 bindings_by_port_id[port_id] = resp.json()['binding']
             else:
+                # In case we actually managed to create the binding,
+                # but something went wrong in between, we can recover
+                if resp.status_code == 409:
+                    error_type = resp.json().get('NeutronError', {}).get('type')
+                    if error_type == 'PortBindingAlreadyExists':
+                        resp = client.get(
+                            '/v2.0/ports/{}/bindings/{}'.format(
```

Comment:
I had a hard time finding that endpoint, as it's not in the networking documentation. I found it here https://specs.openstack.org/openstack/neutron-specs/specs/backlog/pike/portbinding_information_for_nova.html
I think we should put it into the commit-message for future reference.